### PR TITLE
helm: Expose agent DNS proxy parameters as chart values

### PR DESCRIPTION
**Pull request by joaoubaldo**
Previously `extraConfig` and `extraArgs` chart values could be used to customize the agent DNS proxy parameters.
This PR proposes exposing agent DNS proxy parameters as helm chart values. Example:
```
dnsProxy:
  proxyResponseMaxDelay: 200ms
```

Signed-off-by: Joao Ubaldo <me@joaoubaldo.com>

### CODE REVIEW

File: install/kubernetes/cilium/values.yaml
Comment (kaworu) [inline install/kubernetes/cilium/values.yaml]: No need to document the default value in the comment, it is visible one line below and there is a dedicated column in the generated README.

```suggestion
  # -- DNS response code for rejecting DNS requests, available options are '[nameError refused]'.
```

Ditto for the other values introduced.

**Review by kaworu — CHANGES_REQUESTED** [review]
Hi @joaoubaldo :wave: thanks for the patch, I have a small comment but other than that LGTM.

**Comment by joaoubaldo** [comment]
Thanks for reviewing @kaworu 
I've applied the suggestion

**Comment by kaworu** [comment]
Thanks @joaoubaldo :pray: would you mind doing the same for the other values please?

**Comment by joaoubaldo** [comment]
@kaworu of course, sorry I missed that. PR updated.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: Since [the daemon default value is `true`](https://github.com/cilium/cilium/blob/5ba662563e273658086052c27028fa459d84de0e/pkg/defaults/defaults.go#L138-L140), there is now way to disable it from Helm as `--set dnsProxy.enableDnsCompression=false` will omit the option from the ConfigMap and use the daemon's default. Is that a fair statement?
```suggestion
  tofqdns-enable-dns-compression: {{ .Values.dnsProxy.enableDnsCompression | quote }}
```

**Review by kaworu — CHANGES_REQUESTED** [review]
Thanks for the update @joaoubaldo, one more comment.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (joaoubaldo) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: Good catch @kaworu. PR updated. 
For consistency, I've kept the value check before inserting it into the `ConfigMap`
```
{{- if hasKey .Values.dnsProxy "enableDnsCompression" }}
  tofqdns-enable-dns-compression: {{ .Values.dnsProxy.enableDnsCompression | quote }}
{{- end }}```

Comment (kaworu) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: That's reasonable, thanks for the fix!

**Review by kaworu — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

File: install/kubernetes/cilium/values.yaml
Comment (joestringer) [inline install/kubernetes/cilium/values.yaml:2061]: After #20066 was merged, now these changes need to be made to `install/kubernetes/cilium/values.yaml.tmpl` instead, then run `make -C install/kubernetes` to update this file.

**Review by joestringer — CHANGES_REQUESTED** [review]
Thanks for the PR. Just one note, this needs rebasing as the values.yaml has now been templatized. See details below.

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
Hmm, the docs check doesn't seem to like the options in the helm values file in the docs. We can either add all of the new options to the spelling list, or improve the spelling checker to avoid checking the helm values file.

```
Please fix the following spelling mistakes:
* Documentation/helm-reference.rst:372: (dnsProxy)  dnsProxy.dnsRejectResponseCode
* Documentation/helm-reference.rst:372: (dnsRejectResponseCode)  dnsProxy.dnsRejectResponseCode
* Documentation/helm-reference.rst:373: (nameError)  DNS response code for rejecting DNS requests, available options are '[nameError refused]'.
* Documentation/helm-reference.rst:376: (dnsProxy)  dnsProxy.enableDnsCompression
* Documentation/helm-reference.rst:376: (enableDnsCompression)  dnsProxy.enableDnsCompression
* Documentation/helm-reference.rst:380: (dnsProxy)  dnsProxy.endpointMaxIpPerHostname
* Documentation/helm-reference.rst:380: (endpointMaxIpPerHostname)  dnsProxy.endpointMaxIpPerHostname
* Documentation/helm-reference.rst:384: (dnsProxy)  dnsProxy.idleConnectionGracePeriod
* Documentation/helm-reference.rst:384: (idleConnectionGracePeriod)  dnsProxy.idleConnectionGracePeriod
* Documentation/helm-reference.rst:388: (dnsProxy)  dnsProxy.maxDeferredConnectionDeletes
* Documentation/helm-reference.rst:388: (maxDeferredConnectionDeletes)  dnsProxy.maxDeferredConnectionDeletes
* Documentation/helm-reference.rst:392: (dnsProxy)  dnsProxy.minTtl
* Documentation/helm-reference.rst:392: (minTtl)  dnsProxy.minTtl
* Documentation/helm-reference.rst:396: (dnsProxy)  dnsProxy.preCache
* Documentation/helm-reference.rst:396: (preCache)  dnsProxy.preCache
* Documentation/helm-reference.rst:397: (preloaded)  DNS cache data at this path is preloaded on agent startup.
* Documentation/helm-reference.rst:400: (dnsProxy)  dnsProxy.proxyPort
* Documentation/helm-reference.rst:400: (proxyPort)  dnsProxy.proxyPort
* Documentation/helm-reference.rst:404: (dnsProxy)  dnsProxy.proxyResponseMaxDelay
* Documentation/helm-reference.rst:404: (proxyResponseMaxDelay)  dnsProxy.proxyResponseMaxDelay
* Documentation/helm-values.rst:372: (dnsProxy)  dnsProxy.dnsRejectResponseCode
* Documentation/helm-values.rst:372: (dnsRejectResponseCode)  dnsProxy.dnsRejectResponseCode
* Documentation/helm-values.rst:373: (nameError)  DNS response code for rejecting DNS requests, available options are '[nameError refused]'.
* Documentation/helm-values.rst:376: (dnsProxy)  dnsProxy.enableDnsCompression
* Documentation/helm-values.rst:376: (enableDnsCompression)  dnsProxy.enableDnsCompression
* Documentation/helm-values.rst:380: (dnsProxy)  dnsProxy.endpointMaxIpPerHostname
* Documentation/helm-values.rst:380: (endpointMaxIpPerHostname)  dnsProxy.endpointMaxIpPerHostname
* Documentation/helm-values.rst:384: (dnsProxy)  dnsProxy.idleConnectionGracePeriod
* Documentation/helm-values.rst:[38](https://github.com/cilium/cilium/runs/6942605801?check_suite_focus=true#step:5:39)4: (idleConnectionGracePeriod)  dnsProxy.idleConnectionGracePeriod
* Documentation/helm-values.rst:388: (dnsProxy)  dnsProxy.maxDeferredConnectionDeletes
* Documentation/helm-values.rst:388: (maxDeferredConnectionDeletes)  dnsProxy.maxDeferredConnectionDeletes
* Documentation/helm-values.rst:[39](https://github.com/cilium/cilium/runs/6942605801?check_suite_focus=true#step:5:40)2: (dnsProxy)  dnsProxy.minTtl
* Documentation/helm-values.rst:392: (minTtl)  dnsProxy.minTtl
* Documentation/helm-values.rst:396: (dnsProxy)  dnsProxy.preCache
* Documentation/helm-values.rst:396: (preCache)  dnsProxy.preCache
* Documentation/helm-values.rst:397: (preloaded)  DNS cache data at this path is preloaded on agent startup.
* Documentation/helm-values.rst:[40](https://github.com/cilium/cilium/runs/6942605801?check_suite_focus=true#step:5:41)0: (dnsProxy)  dnsProxy.proxyPort
* Documentation/helm-values.rst:400: (proxyPort)  dnsProxy.proxyPort
* Documentation/helm-values.rst:404: (dnsProxy)  dnsProxy.proxyResponseMaxDelay
* Documentation/helm-values.rst:404: (proxyResponseMaxDelay)  dnsProxy.proxyResponseMaxDelay
If the words are not misspelled, run:
Documentation/update-spelling_wordlist.sh dnsProxy dnsRejectResponseCode enableDnsCompression endpointMaxIpPerHostname idleConnectionGracePeriod maxDeferredConnectionDeletes minTtl nameError preCache preloaded proxyPort proxyResponseMaxDelay
```

Could you also squash all the commits together?

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
Thanks, I checked the settings and they seem to match the agent defaults. All the smoke tests passed. Good to merge.

**Comment by joaoubaldo** [comment]
Cool, thanks @joestringer

**Pull request merged**
